test/Kestrel.Core.Tests/KestrelServerTests.cs

```diff
@@ -37,20 +38,17 @@ public void StartWithInvalidAddressThrows()
         }
 
         [Fact]
-        public void StartWithHttpsAddressThrows()
+        public void StartWithHttpsAddressConfiguresHttpsEndpoints()
         {
-            var testLogger = new TestApplicationErrorLogger { ThrowOnCriticalErrors = false };
+            var mockDefaultHttpsProvider = new Mock<IDefaultHttpsProvider>();
 
-            using (var server = CreateServer(new KestrelServerOptions(), testLogger))
+            using (var server = CreateServer(new KestrelServerOptions(), mockDefaultHttpsProvider.Object))
             {
                 server.Features.Get<IServerAddressesFeature>().Addresses.Add("https://127.0.0.1:0");
 
-                var exception = Assert.Throws<InvalidOperationException>(() => StartDummyApplication(server));
+                StartDummyApplication(server);
 
-                Assert.Equal(
-                    $"HTTPS endpoints can only be configured using {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}().",
-                    exception.Message);
-                Assert.Equal(1, testLogger.CriticalErrorsLogged);
+                mockDefaultHttpsProvider.Verify(provider => provider.ConfigureHttps(It.IsAny<ListenOptions>()), Times.Once);
```

Comment:
Not a fan of tests that verify that a certain method was called. Is there a way to build a simple dummy DefaultHttpsProvider and then verify the certificate on the ListenOptions?

Reply:
What is the issue with this kind of test? To test with `DefaultHttpsProvider`, I'd have to make the test project depend on the `Kestrel` project, instead of just `Kestrel.Core`.

Reply:
Hmm. I'll take a bit of a look at it and see if we can make something work. Tests that use mock verification tend to be brittle whenever you change the implementation and they aren't actually testing the desired behavior of the function, they're just testing that the code does what is written. It's a little like a comment over `x += 1` that says `// Adds 1 to x` :). If that's all that we can do, then I suppose it's OK, but I'd like to try and avoid it.